GameApplication/GameApplication/Startup.cs

```diff
@@ -82,11 +87,15 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
                 routes.MapRoute(
                     name: "default",
                     template: "{controller=Games}/{action=Index}/{id?}");
+                
+                routes.MapRoute(
+                    name: "GeneralChat",
+                    template: "{controller=GeneralChat}/{action=Index}");
             });
 
             app.UseSignalR(routes =>
             {
-                routes.MapHub<ChatRoom>("chat");
+                routes.MapHub<ChatRoom>("general_chat");
```

Comment:
You have to update chat.js file. Consider passing chatName to startChat function.